docs/modules/release-notes/pages/0.27.adoc

```diff
@@ -19,34 +21,520 @@ To get started, follow xref:pkl-cli:index.adoc#installation[Installation].#
+    imports {
+      appConfig {
+        sourceModules.add(file("src/main/resources/appConfig.pkl"))
+      }
+    }
+  }
+}
+----
+
+build.gradle.kts::
++
+[source,kotlin]
+----
+pkl {
+  analyzers {
+    imports {
+      register("appConfig") {
+        sourceModules.add(file("src/main/resources/appConfig.pkl"))
+      }
+    }
+  }
+}
+----
+====
+
+For more details, consult the xref:pkl-gradle:index.adoc#analyze-imports[documentation].
+
+==== Tracked file outputs
+
+Tasks created by the Pkl plugin now declare tracked output files (https://github.com/apple/pkl/pull/403[#403]).
+
+This means that downstream tasks do not need to declare an explicit dependency on the Pkl task.
+
+For example, assuming that `evalPkl` is an `EvalTask`:
+
+.build.gradle.kts
+[source,diff]
+----
+ val myGradleTask by tasks.registering {
+   inputs.files(evalPkl)
+-  dependsOn(evalPkl) // <1>
+ }
+----
+<1> No longer necessary to declare this dependency.
+
+[[transitive-modules-computed-by-default]]
+==== `transitiveModules` computed by default 
+
+The `transitiveModules` property of a task is now computed by building the import graph of the source modules (https://github.com/apple/pkl/pull/695[#695]).
+
+This means that Pkl-related tasks no longer need to declare their set of transitive modules, because the pkl-gradle plugin will compute this automatically.
+
+NOTE: This adds latency to each task. To opt out of this behavior, set `transitiveModules` explicitly.
+
+=== Standard library changes
+
+==== Additions to `pkl:base`
+
+New properties and methods have been added to the classes of `pkl:base` (https://github.com/apple/pkl/pull/666[#666], https://github.com/apple/pkl/pull/683[#683]).
+
+New properties and methods have been added to {uri-stdlib-Listing}[`Listing`] and {uri-stdlib-Mapping}[`Mapping`].
+
+One of the goals of this change is to improve the experience of authoring constraints.
+This eliminates the need to convert to collection types as often.
+
+The added properties and methods are:
+
+* {uri-stdlib-Listing}#first[`Listing.first`]
+* {uri-stdlib-Listing}#firstOrNull[`Listing.firstOrNull`]
+* {uri-stdlib-Listing}#last[`Listing.last`]
+* {uri-stdlib-Listing}#lastOrNull[`Listing.lastOrNull`]
+* {uri-stdlib-Listing}#single[`Listing.single`]
+* {uri-stdlib-Listing}#singleOrNull[`Listing.singleOrNull`]
+* {uri-stdlib-Listing}#every()[`Listing.every()`]
+* {uri-stdlib-Listing}#any()[`Listing.any()`]
+* {uri-stdlib-Listing}#contains()[`Listing.contains()`]
+* {uri-stdlib-Mapping}#containsValue()[`Mapping.containsValue()`]
+* {uri-stdlib-Mapping}#every()[`Mapping.every()`]
+* {uri-stdlib-Mapping}#any()[`Mapping.any()`]
+
+With this, the following change can be made to existing constraints:
+
+[source,diff]
+----
+-ipAddresses: Listing<String>(toList().contains("127.0.0.1"))
++ipAddresses: Listing<String>(contains("127.0.0.1"))
+----
+
+Additionally, a new method is added to `String`, called {uri-stdlib-String}#splitLimit()[`String.splitLimit()`].
```

Comment:
It might be getting too granular to do this on per-stdlib-function basis and we already acknowledge in the external readers sections, but shouldn't this also acknowledge contribution?
```suggestion
Additionally, a new method is added to `String`, called {uri-stdlib-String}#splitLimit()[`String.splitLimit()`].
Thanks https://github.com/HT154[@HT154] for contributing this function.
```

Reply:
This is subjective, but my policy was to give acknowledgments for sizable contributions only; @HT154 contributed many other improvements (including bug fixes) that also aren't specifically acknowledged